### PR TITLE
codeintel: Reorganize resolver definitions

### PR DESCRIPTION
**Pull request by efritz**
Roughly breaking `all.go` resolvers into multiple files organized by domain. This is a pre-step to determining how to re-draw the boundary between upload and indexes.

## Test plan

Local GraphQL server started w/o panic.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/codeintel.go [hunk 1]
```diff
@@ -1,9 +1,189 @@
...
-	resolvers.RootResolver
+	resolverstubs.RootResolver
 
 	NodeResolvers() map[string]NodeByIDFunc
 }
+
+//
+//
+
+type Resolver struct {
```
Comment (Numbers88s) [inline cmd/frontend/graphqlbackend/codeintel.go:22]: Why add all these back? 😢

File: enterprise/cmd/frontend/internal/codeintel/init.go [hunk 2]
```diff
@@ -116,7 +117,7 @@ func Init(
 		locationResolverFactory,
 	)
 
-	enterpriseServices.CodeIntelResolver = newResolver(
+	enterpriseServices.CodeIntelResolver = graphqlbackend.NewCodeIntelResolver(
```
Comment (Numbers88s) [inline enterprise/cmd/frontend/internal/codeintel/init.go:120]: Let's not go back 😢

File: cmd/frontend/graphqlbackend/codeintel.go (hunk 1 above)
Comment (efritz) [reply cmd/frontend/graphqlbackend/codeintel.go:22]: It's not back I just moved it into this package.

File: enterprise/cmd/frontend/internal/codeintel/init.go (hunk 2 above)
Comment (efritz) [reply enterprise/cmd/frontend/internal/codeintel/init.go:120]: I have an alternative, stand by!

File: cmd/frontend/graphqlbackend/codeintel.go (hunk 1 above)
Comment (efritz) [reply cmd/frontend/graphqlbackend/codeintel.go:17]: see https://github.com/sourcegraph/sourcegraph/pull/49665#discussion_r1141191549

File: enterprise/cmd/frontend/internal/codeintel/init.go (hunk 2 above)
Comment (efritz) [reply enterprise/cmd/frontend/internal/codeintel/init.go]: Updated! Originally put the root resolver (it's all boilerplate) back into that package due to a dependency on some types defined in graphqlbackend. Figured a way around it - check out the new commit.

**Review by Numbers88s — APPROVED** [review]

File: enterprise/cmd/frontend/internal/codeintel/init.go (hunk 2 above)
Comment (Numbers88s) [reply enterprise/cmd/frontend/internal/codeintel/init.go]: Awesome! Thank you! ✌️

**Pull request merged**